src/bp/core/services/cms.ts

```diff
@@ -230,10 +232,16 @@ export class CMSService implements IDisposeOnExit {
   }
 
   private async _deleteContentElements(botId: string, ids: string[]): Promise<void> {
-    return this.memDb(this.contentTable)
+    const elements = await this.getContentElements(botId, ids)
+    await Promise.map(elements, el => this.moduleLoader.onElementChanged(botId, 'delete', el))
+
+    await this.memDb(this.contentTable)
       .where({ botId })
       .whereIn('id', ids)
       .del()
+
+    const contentTypes = _.uniq(_.map(elements, 'contentType'))
+    await Promise.mapSeries(contentTypes, contentTypeId => this.dumpDataToFile(botId, contentTypeId))
```

Comment:
Why that ?

Reply:
Otherwise, if you delete elements and restart the server, you lose your changes and elements are back. The "dump memdb to json" only occurs during create or updateof element right now